test/pjrt/test_ddp.py

```diff
@@ -31,6 +50,49 @@ def _ddp_init(init_file: str):
   def test_ddp_init(self):
     pjrt.run_multiprocess(self._ddp_init, self.create_tempfile().full_path)
 
+  @staticmethod
+  def _ddp_correctness(init_file: str):
+    rank, world_size = _init_xla_backend(init_file)
+
+    device = xm.xla_device()
+
+    cpu_model = nn.Linear(10, 10)
+    # TODO(@alanwaketan): Investigate whether we can omit the gradient_as_bucket_view option.
+    ddp_model = DDP(copy.deepcopy(cpu_model).to(device), gradient_as_bucket_view=True)
+
+    cpu_optimizer = optim.SGD(cpu_model.parameters(), lr=1e-100)
+    ddp_optimizer = optim.SGD(ddp_model.parameters(), lr=1e-100)
+    loss_fn = nn.MSELoss()
+
+    local_batch_size = 2
+    global_batch_size = local_batch_size * world_size
+    offset = rank * local_batch_size
+    # Lower range probably makes sense too. Anyway, stick to 100 as the original PoC.
+    for step in range(100):
+      # To make torch.randn produce same results across devices.
+      torch.manual_seed(2022 + step)
+
+      cpu_inputs = torch.randn(global_batch_size, 10)
+      cpu_labels = torch.randn(global_batch_size, 10)
+      cpu_loss = _train_step(cpu_model, cpu_inputs, cpu_labels, cpu_optimizer, loss_fn)
+
+      ddp_inputs = copy.deepcopy(cpu_inputs[offset : offset + local_batch_size]).to(device)
+      ddp_labels = copy.deepcopy(cpu_labels[offset : offset + local_batch_size]).to(device)
+      ddp_loss = _train_step(ddp_model, ddp_inputs, ddp_labels, ddp_optimizer, loss_fn)
+      with torch.no_grad():
+        ddp_loss = ddp_loss / world_size
+        dist.all_reduce(ddp_loss)
+
+      # TODO(@alanwaketan): Investigate why the atol here is this low.
+      assert torch.allclose(cpu_loss, ddp_loss, atol=1e-02)
+      _assert_all_close(cpu_model.parameters(), ddp_model.parameters())
+      print(f"iteration {step}: cpu_loss = {cpu_loss}, ddp_loss = {ddp_loss}, cpu_model.parameters() ~= ddp_model.parameters()")
```

Comment:
Will this print 100 times during the unit test? Can you instead print only the final value or guard this behind some verbosity flag?

Reply:
Good point. I was too lazy. Let me add that.